libs/types/src/tokens.rs

```diff
@@ -17,20 +17,27 @@ use cfg_primitives::{
 	Balance, PalletIndex,
 };
 use cfg_traits::{investments::TrancheCurrency as TrancheCurrencyT, HasLocalAssetRepresentation};
-pub use orml_asset_registry::AssetMetadata;
 use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
 use scale_info::TypeInfo;
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
-use sp_runtime::{traits::Get, DispatchError, TokenError};
-use xcm::{
+use sp_runtime::{
+	traits::{ConstU32, Get},
+	DispatchError, TokenError,
+};
+use staging_xcm::{
 	prelude::{AccountKey20, GlobalConsensus, PalletInstance},
 	v3::{MultiLocation, NetworkId},
 	VersionedMultiLocation,
 };
 
 use crate::{domain_address::DomainAddress, xcm::XcmMetadata, EVMChainId};
 
+pub const MAX_ASSET_STRING_LIMIT: u32 = 64;
+
+pub type AssetMetadata =
+	orml_asset_registry::AssetMetadata<Balance, CustomMetadata, ConstU32<MAX_ASSET_STRING_LIMIT>>;
```

Comment:
Now, we need to specify the token names and symbol limits. I guess 64 is enough for all cases.